src/dashboard/components/template-cloud/Sources.js

```diff
@@ -0,0 +1,86 @@
+import { BaseControl, Button, Notice } from '@wordpress/components';
+import { trash } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+const Sources = ({ sourcesData, setSources, isSyncing }) => {
+	const { createNotice } = useDispatch('core/notices');
+	const [ error, setError ] = useState('');
+
+	const deleteSource = (key) => {
+		// eslint-disable-next-line no-alert
+		const confirm = confirm(__('Are you sure you want to delete this source?', 'otter-blocks'));
+
+		if (!confirm) {
+			return;
+		}
+
+		apiFetch({
+			path: `otter/v1/template-cloud/delete-source/${key}`,
+			method: 'DELETE',
+			accept: 'application/json',
+		}).then((response) => {
+			setSources(response.sources);
+			createNotice(
+				'success',
+				__('Source deleted', 'otter-blocks'),
+				{
+					isDismissible: true,
+					type: 'snackbar'
+				}
+			);
+		}).catch((e) => {
+			setError(e?.message ?? __('An unknown error occurred.', 'otter-blocks'));
+		});
+	};
```

Comment:
If you've tested it and it's working then it's fine but in past apiFetch didn't support using DELETE method like this, and I had to do it like this instead:

```
		const response = await apiFetch({
			path: `${ window.hyve.api }/faq/${ id }`,
			method: 'POST',
			headers: {
				'X-HTTP-Method-Override': 'DELETE'
			}
		});
```

In my case, usually the DELETE and POST methods were calling the same API endpoint with different results. I suppose you're using a route that ONLY supports DELETE method and not POST method, it might be working fine.